diff --git a/qb/config.params.sh b/qb/config.params.sh
index 09e89d63af..487bc003a5 100644
--- a/qb/config.params.sh
+++ b/qb/config.params.sh
@@ -101,7 +101,7 @@
 HAVE_PARPORT=auto          # Parallel port joypad support
 HAVE_IMAGEVIEWER=yes       # Built-in image viewer support.
 HAVE_MMAP=auto             # MMAP support
-HAVE_QT=auto               # Qt companion support
+HAVE_QT=no               # Qt companion support
 C89_QT=no
 HAVE_XSHM=no               # XShm video driver support
 HAVE_CHEEVOS=yes           # Retro Achievements
diff --git a/qb/qb.comp.sh b/qb/qb.comp.sh
index a0cced4537..e47d512df0 100644
--- a/qb/qb.comp.sh
+++ b/qb/qb.comp.sh
@@ -83,7 +83,7 @@
 	echo "$echobuf ... $WINDRES"
 fi
 
-if [ "$HAVE_QT" != "no" ]; then
+if [ "$HAVE_QT" = "yes" ]; then
 	echobuf="Checking for moc"
 	if [ -z "$MOC" ]; then
 		MOC="$(exists "moc")" || MOC=""

