--- a/Makefile.common	Wed Jan 23 11:55:14 2019
+++ b/Makefile.common	Wed Jan 23 11:59:25 2019
@@ -3,6 +3,16 @@
 LIBRETRO_COMM_DIR := $(ROOT_DIR)/libretro-common
 WANT_WGL           = 0
 
+ifeq ($(HAVE_SX05RE), 1)
+   DEFINES += -DHAVE_SX05RE
+   HAVE_UPDATE_ASSETS=1
+   HAVE_LIBRETRODB=1
+   HAVE_NETWORKING=1
+   HAVE_LAKKA=1
+   HAVE_ZARCH=0
+   HAVE_QT=0
+   
+endif
 ifeq ($(HAVE_STACK_USAGE), 1)
    DEF_FLAGS += -fstack-usage
 endif
--- a/menu/menu_displaylist.c	2017-08-18 06:30:31.000000000 -0500
+++ b/menu/menu_displaylist.c	2017-08-20 01:54:59.625855817 -0500
@@ -3090,28 +3090,6 @@
    settings_t *settings         = config_get_ptr();
 #endif
 
-#ifdef HAVE_LAKKA
-   menu_entries_append_enum(info->list,
-         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_LAKKA),
-         msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_LAKKA),
-         MENU_ENUM_LABEL_UPDATE_LAKKA,
-         MENU_SETTING_ACTION, 0, 0);
-   count++;
-   menu_entries_append_enum(info->list,
-         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_THUMBNAILS_UPDATER_LIST),
-         msg_hash_to_str(MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST),
-         MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST,
-         MENU_SETTING_ACTION, 0, 0);
-   count++;
-   menu_entries_append_enum(info->list,
-         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_DOWNLOAD_CORE_CONTENT),
-         msg_hash_to_str(MENU_ENUM_LABEL_DOWNLOAD_CORE_CONTENT_DIRS),
-         MENU_ENUM_LABEL_DOWNLOAD_CORE_CONTENT_DIRS,
-         MENU_SETTING_ACTION, 0, 0);
-   count++;
-#elif defined(HAVE_NETWORKING)
-#if defined(__WINRT__) || defined(WINAPI_FAMILY) && WINAPI_FAMILY == WINAPI_FAMILY_PHONE_APP
-#else
    if (settings->bools.menu_show_core_updater)
    {
       menu_entries_append_enum(info->list,
@@ -3121,8 +3099,6 @@
             MENU_SETTING_ACTION, 0, 0);
       count++;
    }
-#endif
-
    menu_entries_append_enum(info->list,
          msg_hash_to_str(MENU_ENUM_LABEL_VALUE_THUMBNAILS_UPDATER_LIST),
          msg_hash_to_str(MENU_ENUM_LABEL_THUMBNAILS_UPDATER_LIST),
@@ -3147,14 +3123,13 @@
       count++;
    }
 
-#ifdef HAVE_UPDATE_ASSETS
    menu_entries_append_enum(info->list,
          msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_ASSETS),
          msg_hash_to_str(MENU_ENUM_LABEL_UPDATE_ASSETS),
          MENU_ENUM_LABEL_UPDATE_ASSETS,
          MENU_SETTING_ACTION, 0, 0);
    count++;
-#endif
+
 
    menu_entries_append_enum(info->list,
          msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_AUTOCONFIG_PROFILES),
@@ -3170,7 +3145,6 @@
          MENU_SETTING_ACTION, 0, 0);
    count++;
 
-#ifdef HAVE_LIBRETRODB
 #if !defined(VITA)
    menu_entries_append_enum(info->list,
          msg_hash_to_str(MENU_ENUM_LABEL_VALUE_UPDATE_DATABASES),
@@ -3178,7 +3152,6 @@
          MENU_ENUM_LABEL_UPDATE_DATABASES,
          MENU_SETTING_ACTION, 0, 0);
    count++;
-#endif
 #endif
 
    menu_entries_append_enum(info->list,
@@ -3217,7 +3190,6 @@
             MENU_SETTING_ACTION, 0, 0);
       count++;
    }
-#endif
 
    return count;
 }

--- a/menu/menu_setting.c	2017-03-12 17:52:42.000000000 -0600
+++ b/menu/menu_setting.c	2017-03-13 19:05:16.526319525 -0600
@@ -4023,23 +4023,6 @@
                &subgroup_info,
                parent_group);
 #endif
-         CONFIG_ACTION(
-               list, list_info,
-               MENU_ENUM_LABEL_REBOOT,
-               MENU_ENUM_LABEL_VALUE_REBOOT,
-               &group_info,
-               &subgroup_info,
-               parent_group);
-         menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_REBOOT);
-
-         CONFIG_ACTION(
-               list, list_info,
-               MENU_ENUM_LABEL_SHUTDOWN,
-               MENU_ENUM_LABEL_VALUE_SHUTDOWN,
-               &group_info,
-               &subgroup_info,
-               parent_group);
-         menu_settings_list_current_add_cmd(list, list_info, CMD_EVENT_SHUTDOWN);
 #endif
 
          CONFIG_ACTION(
@@ -8551,36 +8534,6 @@
                   general_write_handler,
                   general_read_handler,
                   SD_FLAG_NONE);
-
-            CONFIG_BOOL(
-                  list, list_info,
-                  &settings->bools.menu_show_reboot,
-                  MENU_ENUM_LABEL_MENU_SHOW_REBOOT,
-                  MENU_ENUM_LABEL_VALUE_MENU_SHOW_REBOOT,
-                  menu_show_reboot,
-                  MENU_ENUM_LABEL_VALUE_OFF,
-                  MENU_ENUM_LABEL_VALUE_ON,
-                  &group_info,
-                  &subgroup_info,
-                  parent_group,
-                  general_write_handler,
-                  general_read_handler,
-                  SD_FLAG_NONE);
-
-            CONFIG_BOOL(
-                  list, list_info,
-                  &settings->bools.menu_show_shutdown,
-                  MENU_ENUM_LABEL_MENU_SHOW_SHUTDOWN,
-                  MENU_ENUM_LABEL_VALUE_MENU_SHOW_SHUTDOWN,
-                  menu_show_shutdown,
-                  MENU_ENUM_LABEL_VALUE_OFF,
-                  MENU_ENUM_LABEL_VALUE_ON,
-                  &group_info,
-                  &subgroup_info,
-                  parent_group,
-                  general_write_handler,
-                  general_read_handler,
-                  SD_FLAG_NONE);
 #endif
 
 #if defined(HAVE_XMB) || defined(HAVE_OZONE)
@@ -9195,7 +9148,6 @@
                general_read_handler,
                SD_FLAG_NONE);
 
-#if !defined(HAVE_LAKKA)
          CONFIG_BOOL(
                list, list_info,
                &settings->bools.menu_show_core_updater,
@@ -9210,7 +9162,6 @@
                general_write_handler,
                general_read_handler,
                SD_FLAG_NONE);
-#endif
 #endif
 
          CONFIG_BOOL(
