diff -ruN ppsspp-070e29d921845fe324baa4f9c403df0898c2d1ca/SDL/SDLMain.cpp ppsspp-21bdc2b4c95030c2ce7b8608e2772b890da1b000/SDL/SDLMain.cpp
--- ppsspp-070e29d921845fe324baa4f9c403df0898c2d1ca/SDL/SDLMain.cpp	2018-12-06 22:01:16.000000000 -0600
+++ ppsspp-21bdc2b4c95030c2ce7b8608e2772b890da1b000/SDL/SDLMain.cpp	2018-12-06 03:16:54.000000000 -0600
@@ -441,7 +441,7 @@
 	}
 
 	// If we're on mobile, don't try for windowed either.
-#if defined(MOBILE_DEVICE)
+#if defined(USING_GLES2) || defined(MOBILE_DEVICE)
     mode |= SDL_WINDOW_FULLSCREEN;
 #else
     mode |= SDL_WINDOW_RESIZABLE;
