--- a/CMakeLists.txt	2018-02-26 00:08:30.150219106 +0800
+++ b/CMakeLists.txt	2018-02-26 00:09:16.474069170 +0800
@@ -96,6 +96,7 @@
 # :: Processors
 option(ARMV7 "Set to ON if targeting an ARMv7 processor" ${ARMV7})
 option(ARM "Set to ON if targeting an ARM processor" ${ARM})
+option(ARM64 "Set to ON if targeting an ARM64 processor" ${ARM64})
 option(MIPS "Set to ON if targeting a MIPS processor" ${MIPS})
 option(X86 "Set to ON if targeting an X86 processor" ${X86})
 option(X86_64 "Set to ON if targeting an X86_64 processor" ${X86_64})
@@ -117,6 +118,10 @@
 option(USE_WAYLAND_WSI "Set to ON to require Wayland support for Vulkan" ${USE_WAYLAND_WSI})
 option(USE_ADDRESS_SANITIZER "Use Clang memory sanitizer" ${USE_ADDRESS_SANITIZER})
 
+if(ARM64 OR ARMV7 OR ARM)
+	set(VULKAN OFF)
+endif()
+
 if(UNIX AND NOT (APPLE OR ANDROID) AND VULKAN)
 	add_definitions(-DVK_USE_PLATFORM_XLIB_KHR)
 	# add_definitions(-DVK_USE_PLATFORM_XCB_K

