From 6e123ba8c60897a6f7e9dff8cb1f606576ab859c Mon Sep 17 00:00:00 2001
From: Cristi Mitrana <mitu+retropie@fastmail.fm>
Date: Thu, 2 May 2019 11:54:55 +0300
Subject: [PATCH] scraper: change ScreenScraper API URL

---
 es-app/src/scrapers/ScreenScraper.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/es-app/src/scrapers/ScreenScraper.h b/es-app/src/scrapers/ScreenScraper.h
index 8cdfd216a..4c749008f 100644
--- a/es-app/src/scrapers/ScreenScraper.h
+++ b/es-app/src/scrapers/ScreenScraper.h
@@ -27,7 +27,7 @@ class ScreenScraperRequest : public ScraperHttpRequest
 		const std::string API_DEV_U = { 91, 32, 7, 17 };
 		const std::string API_DEV_P = { 108, 28, 54, 55, 83, 43, 91, 44, 30, 22, 41, 12, 0, 108, 38, 29 };
 		const std::string API_DEV_KEY = { 54, 73, 115, 100, 101, 67, 111, 107, 79, 66, 68, 66, 67, 56, 118, 77, 54, 88, 101, 54 };
-		const std::string API_URL_BASE = "https://screenscraper.fr/api2";
+		const std::string API_URL_BASE = "https://www.screenscraper.fr/api2";
 		const std::string API_SOFT_NAME = "Emulationstation " + static_cast<std::string>(PROGRAM_VERSION_STRING);
 
 		/** Which type of image artwork we need. Possible values (not a comprehensive list):
