--- a/es-core/src/guis/GuiInputConfig.cpp
+++ b/es-core/src/guis/GuiInputConfig.cpp
@@ -349,10 +349,11 @@
 		// digital triggers are unwanted
 		if (input.type == TYPE_BUTTON && (input.id == 6 || input.id == 7))
 			return true;
-		// ignore analog values < 0
+	}
+	// ignore analog values < 0
 		if (input.type == TYPE_AXIS && (input.id == 2 || input.id == 5) && input.value < 0)
 			return true;
-	}
+	
 #endif
 
 	return false;
